### ISSUE TITLE
Insufficient permissions of ruby files in gapic-common >=0.4.1 

### ISSUE THREAD

**Issue by grobie** [googleapis/gapic-generator-ruby#599]
#### Environment details

  - Programming language: Ruby
  - OS: Linux
  - Language runtime version: 3.0.1p64
  - Package version: gapic-common 0.4.1 / 0.4.2

#### Steps to reproduce

1. Add `gapic-common` to bundler Gemfile
2. `bundle install` system wide with sudo permissions (under `/usr/lib/ruby/gems/3.0.0/`)
3. Try loading anything using gapic-common

   ```console
   $ bundle exec ruby -r gapic-common -e "puts Gapic::Common"
   <internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:160:in `require': cannot load such file -- gapic-common (LoadError)
           from <internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:160:in `rescue in require'
           from <internal:/usr/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:149:in `require'
   ```

#### Root cause

The issue is that the all files in the gapic-common releases since version 0.4.1 have the file permissions 640 instead of the typical 644, forbidding world users to read the library files:

```console
$ find /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.* -type f -name '*.rb' -printf '%m %p\n'
644 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.0/lib/gapic-common.rb
644 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.0/lib/gapic/rest.rb
644 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.0/lib/gapic/stream_input.rb
644 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.0/lib/gapic/config/method.rb
644 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.0/lib/gapic/common/version.rb
644 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.0/lib/gapic/operation/retry_policy.rb
644 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.0/lib/gapic/headers.rb
644 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.0/lib/gapic/call_options/retry_policy.rb
644 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.0/lib/gapic/grpc.rb
644 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.0/lib/gapic/common.rb
644 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.0/lib/gapic/protobuf.rb
644 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.0/lib/gapic/config.rb
644 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.0/lib/gapic/paged_enumerable.rb
644 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.0/lib/gapic/rest/error.rb
644 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.0/lib/gapic/rest/faraday_middleware.rb
644 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.0/lib/gapic/rest/client_stub.rb
644 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.0/lib/gapic/operation.rb
644 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.0/lib/gapic/call_options.rb
644 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.0/lib/gapic/grpc/service_stub.rb
644 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.0/lib/gapic/grpc/status_details.rb
644 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.0/lib/gapic/grpc/service_stub/rpc_call.rb
640 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.2/lib/gapic-common.rb
640 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.2/lib/gapic/rest.rb
640 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.2/lib/gapic/stream_input.rb
640 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.2/lib/gapic/config/method.rb
640 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.2/lib/gapic/common/version.rb
640 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.2/lib/gapic/operation/retry_policy.rb
640 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.2/lib/gapic/headers.rb
640 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.2/lib/gapic/call_options/retry_policy.rb
640 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.2/lib/gapic/grpc.rb
640 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.2/lib/gapic/common.rb
640 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.2/lib/gapic/protobuf.rb
640 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.2/lib/gapic/config.rb
640 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.2/lib/gapic/paged_enumerable.rb
640 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.2/lib/gapic/rest/error.rb
640 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.2/lib/gapic/rest/faraday_middleware.rb
640 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.2/lib/gapic/rest/client_stub.rb
640 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.2/lib/gapic/operation.rb
640 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.2/lib/gapic/call_options.rb
640 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.2/lib/gapic/grpc/service_stub.rb
640 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.2/lib/gapic/grpc/status_details.rb
640 /usr/lib/ruby/gems/3.0.0/gems/gapic-common-0.4.2/lib/gapic/grpc/service_stub/rpc_call.rb
```

Of all ruby gem files, only gapic-common uses 640:

```console
$ find /usr/lib/ruby/gems/3.0.0/gems/ -type f -name '*.rb' -printf '%m\n' | sort | uniq -c
     22 444
     21 640
   7437 644
     44 755
```

#### Workaround

```console
$ sudo find $(bundle info gapic-common --path) -type f -name '*.rb' -exec chmod 644 '{}' +
```

**Comment by dazuma** [comment]
Oh shoot, thanks for letting us know. Will get a fixed release out momentarily.

**Comment by dazuma** [comment]
Fixed in [gapic-common 0.4.3](https://rubygems.org/gems/gapic-common/versions/0.4.3). Thanks for the report, and sorry for the hassle.
